### DISCUSSION TITLE
Does this rule https://azure.github.io/PSRule.Rules.Azure/en/rules/Azure.Template.UseParameters/ test parameters from the parameter file

### DISCUSSION THREAD

**Discussion by JonasCordsen** [Azure/PSRule.Rules.Azure#2465] [Q&A] [answered]
I am testing if my parameter file has missing or too many parameters, so that if I have a parameter that is not specified in the bicep file, I will get a warning in PSRule

It seems like this might be the test, but it does not catch it, am I reading the rule wrong, or is there someting els going on?

**Comment by BernieWhite** [comment, marked as answer]
@JonasCordsen If a parameter file is missing requires parameters expand will flag an error.

A parameter file with too many parameters is not flagged today. But raised as an issue here:  #1993

This rule specifically, flag if a parameter is defined but not used. It would apply if you defined a template or Bicep code with parameters but then did not reference those parameters anywhere. It doesn't apply to parameter files today.
